### PR TITLE
Pattern Assembler - Remove feature flags in preparation to add categories

### PR DESCRIPTION
**Pull request by miksansegundo**
Related to https://github.com/Automattic/wp-calypso/pull/72144 https://github.com/Automattic/wp-calypso/pull/74060

## Proposed Changes

In preparation for adding the pattern categories, let's remove the flags that are or will be enabled. 

* Remove pattern-assembler/write-flow 
* Remove pattern-assembler/sidebar-revamp

## Testing Instructions

* Click on the Calypso live link on the first comment
* Create a new site
* On the goals screen, choose `Write and publish`
* Continue until the Design Picker and scroll down to click `Start designing`
* Verify that the new sidebar appears and works as expected. See pdtkmj-1eO-p2 
* Click `Create your homepage` and `[ + ]` Add patterns
* Check that posts patterns are in the first position
* Go back to the goals screen and don't choose any goal
* Verify that the posts patterns aren't in the first position 
* Continue to verify that patterns are added to a Home template on your site

## Pre-merge Checklist

- [X] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [X] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [X] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [X] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [X] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [X] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by miksansegundo** [comment]
@Automattic/lego This PR removes the flag for the new sidebar and should be merged after the new sidebar is enabled on production.

**Review by arthur791004 — COMMENTED** [review]
Looks good to me. There is a type error on the unit test. We have to fix it before merging

> This PR removes the flag for the new sidebar and should be merged after the new sidebar is enabled on production.

[nit] If we remove the feature flag of the "write" flow in another PR, then we can merge that immediately 🙂

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-layout.tsx [hunk 1]
```diff
@@ -24,120 +23,17 @@ type PatternLayoutProps = {
...
-							{ sections.length > 0 &&
-								sections.map( ( { category, key }: Pattern, index ) => {
-									return (
-										<m.li
-											key={ key }
-											layout="position"
-											exit={ { opacity: 0, x: -50, transition: { duration: 0.2 } } }
-											className="pattern-layout__list-item"
-										>
-											<Icon className="pattern-layout__icon" icon={ group } size={ 24 } />
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-layout.tsx:86]: We can also remove the style of this class as it's un-used

**Comment by arthur791004** [comment]
Let's rebase onto the trunk and move it forward 🙂

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/patterns-data.ts [hunk 2]
```diff
@@ -365,13 +365,13 @@ const useSectionPatterns = () => {
 			},
 			{
 				id: 6309,
-				name: '6309',
+				name: 'Names List',
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/patterns-data.ts:368]: 😂

**Review by arthur791004 — APPROVED** [review]
LGTM 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7879721

Hi @miksansegundo, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Replace a pattern`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7879721).

**Pull request merged**